browser-sign-in/app/src/main/java/com/okta/android/samples/browser_sign_in/data/UserDashboardViewModel.kt

```diff
@@ -0,0 +1,148 @@
+import timber.log.Timber
+
+class UserDashboardViewModel : ViewModel() {
+    private val _loginState = MutableLiveData<LoginState>(LoginState.Loading)
+    private val _userGreetingState = MutableLiveData<UserGreetingState>(UserGreetingState.Loading)
+    private val _userInfoState = MutableLiveData<UserInfoState>(UserInfoState.Empty)
+
+    val loginState: LiveData<LoginState> = _loginState
+    val userGreetingState: LiveData<UserGreetingState> = _userGreetingState
+    val userInfoState: LiveData<UserInfoState> = _userInfoState
+
+    init {
+        viewModelScope.launch {
+            refreshCurrentLoginState()
+        }
+    }
+
+    fun logoutOfBrowser(context: Context) {
+        viewModelScope.launch {
+            _loginState.value = LoginState.Loading
+
+            val result =
+                CredentialBootstrap.oidcClient.createWebAuthenticationClient().logoutOfBrowser(
+                    context = context,
+                    redirectUrl = BuildConfig.SIGN_OUT_REDIRECT_URI,
+                    CredentialBootstrap.defaultCredential().token?.idToken ?: "",
+                )
+            when (result) {
+                is OidcClientResult.Error -> {
+                    Timber.e(result.exception, "Failed to logout.")
+                    _loginState.value = LoginState.LogoutFailed(
+                        errorMessage = "Failed to logout.",
+                    )
+                }
+                is OidcClientResult.Success -> {
+                    CredentialBootstrap.defaultCredential().delete()
+                    _loginState.value = LoginState.LoggedOut
+                }
+            }
+        }
+    }
+
+    fun fetchUserInfo() {
+        viewModelScope.launch {
+            _userInfoState.value = UserInfoState.Loading
+            val credential = CredentialBootstrap.defaultCredential()
+
+            when (val result = credential.getUserInfo()) {
+                is OidcClientResult.Error -> {
+                    Timber.e(result.exception, "Failed to fetch userinfo.")
+                    _userInfoState.value = UserInfoState.Loaded(
+                        userInfoText = "Failed to fetch userinfo.",
+                    )
+                }
+                is OidcClientResult.Success -> {
+                    val userClaimsList =
+                        result.result.deserializeClaims(JsonObject.serializer()).asList()
+                    _userInfoState.value = UserInfoState.Loaded(
+                        userInfoText = userClaimsList.joinToString("\n"),
+                    )
+                }
+            }
+        }
+    }
+
+    private suspend fun refreshCurrentLoginState() {
+        val credential = CredentialBootstrap.defaultCredential()
+        if (credential.token == null) {
+            _userGreetingState.value = UserGreetingState.Loading
+            _loginState.value = LoginState.LoggedOut
+        } else {
+            val userName = credential.idToken()?.name ?: ""
```

Comment:
Might be worth extracting `credential.idToken()` into a variable. Calling the `idToken` method has to do some parsing. Although it isn't a huge deal, I'd call it best practice.

That being said. It might be better to key your logged out state off of the presence of that non null id token, rather than just having a non null token. Having an id token is what identifies a user. We don't _really_ support having a null id token. (It's possible that it works, but I've yet to see a Open ID Provider that does it and works with the SDK.)